### ISSUE TITLE
Issue Chapter5.1: 6144 is from vocab_size in the linear layer for output instead of "输入的形状"

### ISSUE THREAD

**Issue by yonggui-yan** [datawhalechina/happy-llm#99]
### 1. 遇到问题的章节 / Affected Chapter

Chapter5.1

### 2. 具体问题描述 / Problem Description

The following sentence is not accurate. The shape of input x is (1, 50).  6144 is parameter of randint, i.e., the vocab_size. 
Output size [1, 1, 6144] is not the same as input size (1, 50). 

"同样大家可以使用下面的代码来对Transformer模块进行测试，可以看到代码**最终输出的形状为torch.Size([1, 1, 6144])**，**与我们输入的形状一致**，说明模块的实现是正确的。
```python
# LLaMA2Model.forward 接受两个参数，tokens和targets，其中tokens是输入的张量, 应为int类型
x = torch.randint(0, 6144, (1, 50)) # [bs, seq_len]
# 实例化LLaMA2Model
model = Transformer(args=args)
# 计算model的全部参数
num_params = sum(p.numel() for p in model.parameters())
print('Number of parameters:', num_params)

out = model(x)
print(out.logits.shape) # [batch_size, 1, vocab_size]
```
"

### 3. 问题重现材料 / Reproduction Materials

change the sentence cited in "Problem Description".

### 确认事项 / Verification

- [x] 此问题未在过往Issue中被报告过 / This issue hasn't been reported before

**Comment by KMnO4-zx** [comment]
这里输出的是logits.shape，应该为[batch_size, 1, vocab_size]，所以[1, 1, 6144]是没有错误的。如下代码所示,`logits = self.output(h[:, [-1], :])`

```python
def forward(self, tokens: torch.Tensor, targets: Optional[torch.Tensor] = None, **keyargs) -> torch.Tensor:
        """
        - tokens: Optional[torch.Tensor], 输入 token 张量。
        - targets: Optional[torch.Tensor], 目标 token 张量。
        - kv_cache: bool, 是否使用键值缓存。
        - keyargs: 其他关键字参数。

        - self.OUT: CausalLMOutputWithPast, 包含 logits 和损失。
        """

        if 'input_ids' in keyargs:
            tokens = keyargs['input_ids']
        if 'attention_mask' in keyargs:
            targets = keyargs['attention_mask']

        # 前向传播函数
        _bsz, seqlen = tokens.shape
        # 通过词嵌入层和Dropout层
        h = self.tok_embeddings(tokens)
        h = self.dropout(h)
        # 获取相对位置嵌入的频率
        freqs_cos = self.freqs_cos[:seqlen]
        freqs_sin = self.freqs_sin[:seqlen]

        # 通过Decoder层
        for layer in self.layers:
            h = layer(h, freqs_cos, freqs_sin)
        # 通过归一化层
        h = self.norm(h)

        if targets is not None:
            # 如果给定了目标，计算损失
            logits = self.output(h)
            self.last_loss = F.cross_entropy(logits.view(-1, logits.size(-1)), targets.view(-1), ignore_index=0, reduction='none')
        else:
            # 推理时的小优化：只对最后一个位置的输出进行前向传播
            logits = self.output(h[:, [-1], :]) 
            self.last_loss = None

        # 设置输出
        self.OUT.__setitem__('logits', logits)
        self.OUT.__setitem__('last_loss', self.last_loss)
        return self.OUT
```
